packages/contracts-bedrock/src/L2/interfaces/ICrossL2Inbox.sol

```diff
@@ -1,53 +1,70 @@
+    uint256 logIndex;
+    uint256 timestamp;
+    uint256 chainId;
+}
+
 /// @title ICrossL2Inbox
 /// @notice Interface for the CrossL2Inbox contract.
 interface ICrossL2Inbox {
-    /// @notice The struct for a pointer to a message payload in a remote (or local) chain.
-    struct Identifier {
-        address origin;
-        uint256 blockNumber;
-        uint256 logIndex;
-        uint256 timestamp;
-        uint256 chainId;
-    }
+    error ReentrantCall();
+
+    /// @notice Thrown when the caller is not DEPOSITOR_ACCOUNT when calling `setInteropStart()`
+    error NotDepositor();
+
+    /// @notice Thrown when attempting to set interop start when it's already set.
+    error InteropStartAlreadySet();
+
+    /// @notice Thrown when a non-written transient storage slot is attempted to be read from.
+    error NotEntered();
+
+    /// @notice Thrown when trying to execute a cross chain message with an invalid Identifier timestamp.
+    error InvalidTimestamp();
+
+    /// @notice Thrown when trying to execute a cross chain message with an invalid Identifier chain ID.
+    error InvalidChainId();
+
+    /// @notice Thrown when trying to execute a cross chain message and the target call fails.
+    error TargetCallFailed();
+
+    /// @notice Thrown when trying to execute a cross chain message on a deposit transaction.
+    error NoExecutingDeposits();
+
+    event ExecutingMessage(bytes32 indexed msgHash, Identifier id);
+
+    function version() external view returns (string memory);
 
     /// @notice Returns the interop start timestamp.
     /// @return interopStart_ interop start timestamp.
     function interopStart() external view returns (uint256 interopStart_);
 
     /// @notice Returns the origin address of the Identifier.
-    /// @return origin_ The origin address of the Identifier.
-    function origin() external view returns (address origin_);
+    function origin() external view returns (address);
 
     /// @notice Returns the block number of the Identifier.
-    /// @return blockNumber_ The block number of the Identifier.
-    function blockNumber() external view returns (uint256 blockNumber_);
+    function blockNumber() external view returns (uint256);
 
     /// @notice Returns the log index of the Identifier.
-    /// @return logIndex_ The log index of the Identifier.
-    function logIndex() external view returns (uint256 logIndex_);
+    function logIndex() external view returns (uint256);
 
     /// @notice Returns the timestamp of the Identifier.
-    /// @return timestamp_ The timestamp of the Identifier.
-    function timestamp() external view returns (uint256 timestamp_);
+    function timestamp() external view returns (uint256);
 
     /// @notice Returns the chain ID of the Identifier.
-    /// @return chainId_ The chain ID of the Identifier.
-    function chainId() external view returns (uint256 chainId_);
+    function chainId() external view returns (uint256);
+
+    function setInteropStart() external;
```

Comment:
no commets / natspec here?

Reply:
The natspec are only required in the contract itself. Also the intention in the near future is for interfaces to be auto generated